From b1c7c72399f821915d280e552b1d719215d13e4c Mon Sep 17 00:00:00 2001
From: Kranthi G <gudishax.kranthikumar@intel.com>
Date: Wed, 18 Nov 2015 15:26:17 +0530
Subject: [PATCH 3924/4301] ASoC: Intel: Skylake: Add updown mixer changes

commit 040a666f285de6dcfd1c327c5f14c4cc1285a675 from
git://git@github.com:01org/linux-intel-4.9.git

As firmware structure modified for updown mixer,
this commit will add channel map member to updown
mixer structure and also changes the size of coefficints to 8.

Change-Id: Ie825def1852a5d2d81e7bf5b1e5d9b2138362211
Signed-off-by: Kranthi G <gudishax.kranthikumar@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/7534
Reviewed-by: Babu, Ramesh <ramesh.babu@intel.com>
Tested-by: Babu, Ramesh <ramesh.babu@intel.com>
---
 sound/soc/intel/skylake/skl-messages.c |    3 ++-
 sound/soc/intel/skylake/skl-topology.h |    3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/sound/soc/intel/skylake/skl-messages.c b/sound/soc/intel/skylake/skl-messages.c
index 5d1bf68..dee4bc4 100644
--- a/sound/soc/intel/skylake/skl-messages.c
+++ b/sound/soc/intel/skylake/skl-messages.c
@@ -645,10 +645,11 @@ static void skl_set_updown_mixer_format(struct skl_sst *ctx,
 
 	/* Select F/W default coefficient */
 	mixer_mconfig->coeff_sel = 0x0;
+	mixer_mconfig->ch_map = fmt->ch_map;
 
 	/* User coeff, don't care since we are selecting F/W defaults */
 	for (i = 0; i < UP_DOWN_MIXER_MAX_COEFF; i++)
-		mixer_mconfig->coeff[i] = 0xDEADBEEF;
+		mixer_mconfig->coeff[i] = 0x0;
 }
 
 /*
diff --git a/sound/soc/intel/skylake/skl-topology.h b/sound/soc/intel/skylake/skl-topology.h
index c42e30c..158c96a 100644
--- a/sound/soc/intel/skylake/skl-topology.h
+++ b/sound/soc/intel/skylake/skl-topology.h
@@ -34,7 +34,7 @@
 #define MAX_FIXED_DMIC_PARAMS_SIZE 727
 
 /* Maximum number of coefficients up down mixer module */
-#define UP_DOWN_MIXER_MAX_COEFF		6
+#define UP_DOWN_MIXER_MAX_COEFF		8
 
 #define MODULE_MAX_IN_PINS	8
 #define MODULE_MAX_OUT_PINS	8
@@ -149,6 +149,7 @@ struct skl_up_down_mixer_cfg {
 	u32 coeff_sel;
 	/* Pass the user coeff in this array */
 	s32 coeff[UP_DOWN_MIXER_MAX_COEFF];
+	u32 ch_map;
 } __packed;
 
 struct skl_algo_cfg {
-- 
1.7.5.4

